src/util/misc.ts

```diff
@@ -369,4 +370,37 @@ export function isBitmapImageMimeType(mimeType: string): boolean {
 
 export function isWebpCandidateImageMimeType(webp: boolean, content_type: string) {
   return webp && WEBP_CANDIDATE_IMAGE_MIME_TYPE.test(content_type);
-}
\ No newline at end of file
+}
+
+/*
+ * Get best fitting MIME type from contentType or pathname
+ * Preference:
+ *   1. extension with image meme-type
+ *   2. any contentType
+ *   3. any extension type
+ *   4. null
+ */
+export function getMimeType(contentType: string, url: string): string {
+  let pMimeType: string;
+  if (url) {
+    let pathname = new URL(url).pathname;
+    pMimeType = mime.lookup(pathname);
+    if (!pMimeType && pathname.indexOf('/revision') !== -1) {
+      pMimeType = mime.lookup(
+        pathname.slice(0, pathname.indexOf('/revision')),
+      )
+    }
+    if (pMimeType && pMimeType.startsWith('image/')) {
```

Comment:
Many Wikipedians make errors in image extensions, ie. use a png for a joeg picture, etc. But mime-type will still be OK IMO. I believe this filter is too large. I would apply only if:
* given contentType mainpart is not « image »
* I would consider « audio » and « video » cases as well

Trusting the extension over the given mimeType should be the exception. Not the rule.